drivers/gpio/gpio_sam0.c

```diff
@@ -4,48 +4,132 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <errno.h>
-#include <kernel.h>
-#include <device.h>
-#include <init.h>
-#include <soc.h>
 #include <gpio.h>
+#include <soc.h>
+
+#include "gpio_utils.h"
 
 struct gpio_sam0_config {
 	PortGroup *regs;
 };
 
-#define DEV_CFG(dev) \
+struct gpio_sam0_data {
+	sys_slist_t cb;
+	u32_t cb_pins;
+};
+
+DEVICE_DECLARE(gpio_sam0_0);
```

Comment:
This line should be placed in the device section at the bottom part of the file.

Reply:
Done.